kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/richmessaging/FaqAlRichMessage.java

```diff
@@ -0,0 +1,98 @@
+package com.applozic.mobicomkit.uiwidgets.conversation.richmessaging;
+
+import android.content.Context;
+import android.text.TextUtils;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import com.applozic.mobicomkit.api.conversation.Message;
+import com.applozic.mobicomkit.uiwidgets.AlCustomizationSettings;
+import com.applozic.mobicomkit.uiwidgets.R;
+import com.applozic.mobicomkit.uiwidgets.conversation.richmessaging.callbacks.ALRichMessageListener;
+import com.applozic.mobicomkit.uiwidgets.conversation.richmessaging.models.ALRichMessageModel;
+import com.applozic.mobicommons.json.GsonUtils;
+
+import java.util.List;
+
+public class FaqAlRichMessage extends AlRichMessage {
+
+    public FaqAlRichMessage(Context context, LinearLayout containerView, Message message, ALRichMessageListener listener, AlCustomizationSettings alCustomizationSettings) {
+        super(context, containerView, message, listener, alCustomizationSettings);
+    }
+
+    @Override
+    public void createRichMessage() {
+        super.createRichMessage();
+        LinearLayout faqReplyLayout = containerView.findViewById(R.id.alFaqReplyLayout);
+        LinearLayout faqLayout = containerView.findViewById(R.id.alFaqLayout);
+        setupAlRichMessage(faqLayout, faqReplyLayout, model);
+    }
+
+    @Override
+    protected void setupAlRichMessage(ViewGroup faqLayout, ViewGroup faqReplyLayout, ALRichMessageModel model) {
+        super.setupAlRichMessage(faqLayout, faqReplyLayout, model);
+        if(model != null) {
+            TextView headerText = faqLayout.findViewById(R.id.headerText);
+            TextView titleText = faqLayout.findViewById(R.id.questionText);
+            TextView descriptionText = faqLayout.findViewById(R.id.bodyText);
+            TextView buttonLabel = faqReplyLayout.findViewById(R.id.buttonLabel);
+            TextView actionYes = faqReplyLayout.findViewById(R.id.actionYes);
+            TextView actionNo = faqReplyLayout.findViewById(R.id.actionNo);
+
+            if (model.getPayload() != null) {
+                ALRichMessageModel.ALPayloadModel payload = (ALRichMessageModel.ALPayloadModel) GsonUtils.getObjectFromJson(model.getPayload(), ALRichMessageModel.ALPayloadModel.class);
+                if (payload != null) {
+                    if (!TextUtils.isEmpty(payload.getTitle())) {
+                        titleText.setVisibility(View.VISIBLE);
+                        titleText.setText(getHtmlText(payload.getTitle()));
+                    } else {
+                        titleText.setVisibility(View.GONE);
```

Comment:
can this be the initial state so that else and setting go GONE is not required anywhere

Reply:
done